registrations/api.py

```diff
@@ -277,36 +282,28 @@ def filter_queryset(self, queryset):
     def filter_registration(self, queryset, name, value: list[str]):
         user = self.request.user
 
-        # Get admin organizations and descendants only once instead of using is_admin method
-        registration_admin_organizations = (
-            user.get_registration_admin_organizations_and_descendants()
-        )
+        try:
+            registrations = Registration.objects.filter(pk__in=value)
+        except ValueError:
+            registrations = Registration.objects.none()
 
-        registrations = []
-        for pk in value:
-            try:
-                registration = Registration.objects.select_related(
-                    "event__publisher"
-                ).get(pk=pk)
-            except (ValueError, Registration.DoesNotExist):
-                raise ParseError(
-                    _("Registration with id {pk} doesn't exist.").format(pk=pk)
-                )
+        if not registrations:
+            raise ParseError(_("No registrations found with the given IDs."))
```

Comment:
Yeaah LE is using ParseError so much, using it qualifies as a code convention at this point, but if we're pedantic, it's meant for situations where the request is malformed and cannot be parsed (like invalid JSON). 

ValidationError might be more appropriate, like 

```python
raise ValidationError({
    "registration": _("No registrations found with the given IDs.")
})
 ```

Double checking that raising an error explicitly is really preferred here over just returning empty, `"count": 0`, page? If yes, should remember to document it in the swagger (if these endpoints are added there ?).

Reply:
There is actually no proper reason for not returning an empty qs instead of raising the exception other than that I didn't want to touch the existing logic too much. But this makes a lot of sense so I changed this to return the empty qs.